### PR TITLE
1038-passing-kwargs-when-saving-an-attachment

### PR DESCRIPTION
**Pull request by raduiordache**
Fixes #<issue>

## Checklist:
- [x] My code follows the pykechain style
    - [x] I added type hinting to all functions
    - [x] I added documentation for all public functions
    - [x] I locally used `tox` to check for dists and docs errors
    - [x] My code passes the `pep8` and `flake8` linting checks and no warnings
    - [x] I removed unused imports, using `Code > Optimize Imports` in PyCharm
    - [x] I used [`black`](https://github.com/psf/black) to format the new code
- [x] I have added tests that prove my fix is effective or that my feature works
    - [x] I committed fresh test cassettes
    - [x] I have proper test coverage (don't decline the coverage of the test)
- [x] I asked another teammate to review the code
- [x] I update the Changelog.md with the appropriate changes.

### CODE REVIEW

**Comment by raduiordache** [comment]
I've also written a test for it. However, while the test passes succesfully when tested on the live instance, it is failing when the cassettes are generated. It appears that something is going wrong in the `save_as()`, but I cannot figure out what. Do you guys have any suggestions?

File: tests/models/test_property_attachment.py [hunk 1]
```diff
@@ -98,3 +104,27 @@ def test_add_with_properties(self):
...
+    #
+    #     with temp_chdir() as target_dir:
+    #         filepath_xl = os.path.join(target_dir, 'image_xl.png')
+    #         self.property.save_as(filename=filepath_xl, size='XL')
+    #         path = os.path.join(target_dir, filepath_xl)
+    #         image_xl = Image.open(path)
+    #         self.assertEqual(image_xl.width, 2048)
+    #
+    #         filepath_sqs = os.path.join(target_dir, 'image_sqs.png')
+    #         self.property.save_as(filename=filepath_sqs, size='SQS')
```
Comment (jberends) [inline tests/models/test_property_attachment.py:125]: Would it be nice to shadow the enums from KE-chain here?
It also could be wise to reduce the image size that you are uploading for the sake of the tests.
The SQS is the same as a `size=h,w` so we could inject size=1,1 or something so we can upload a 4x4 image for testing :)

File: requirements.txt [hunk 2]
```diff
@@ -15,6 +15,7 @@ betamax==0.8.1
 coverage==6.3.2
 pytest-cov==3.0.0
 coveralls==3.3.1
+Pillow==5.2.0
```
Comment (jberends) [inline requirements.txt:18]: This is needed for CI right? Not as a requirement for pykechain. If not we need also to update setup.py

**Review by jberends — CHANGES_REQUESTED** [review]
Some minor review items for discussion - in general it is correct. Suggestion to reduce the image size test. Tests are also suppressed.

File: tests/models/test_property_attachment.py (hunk 1 above)
Comment (jberends) [reply tests/models/test_property_attachment.py:125]:
```ATTACHMENT_IMAGE_SIZES = dict(
    SQXS=(100, 100),  # pixels square
    XS=(100,),  # pixels width
    S=(320,),
    SQS=(320, 320),
    M=(640,),
    SQM=(640, 640),
    L=(1024,),
    SQL=(1024, 1024),
    XL=(2048,),
    SQXL=(2048, 2048),
    XXL=(4096,),
    SQXXL=(4096, 4096)
)```

Comment (jberends) [reply tests/models/test_property_attachment.py:125]: or a 200 x 200 image and request a SQXS image (=(100, 100))

File: requirements.txt (hunk 2 above)
Comment (raduiordache) [reply requirements.txt:18]: That's right, will move it to the right requirements file

File: tests/models/test_property_attachment.py (hunk 1 above)
Comment (raduiordache) [reply tests/models/test_property_attachment.py]: Added a smaller image and adapted the test. Test still fails when recording cassettes, hence it is skipped until we can figure out why it's failing.

**Comment by jberends** [comment]
> I've also written a test for it. However, while the test passes succesfully when tested on the live instance, it is failing when the cassettes are generated. It appears that something is going wrong in the `save_as()`, but I cannot figure out what. Do you guys have any suggestions?

Not really. But I am ok to skip this in CI as long as it tests possively local. We have that in more places.

**Review by jberends — CHANGES_REQUESTED** [review]
- [x] add changelog
- [ ] make sure tests pass succesfully

**Review by jberends — APPROVED** [review]

**Pull request merged**